lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+          manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
+          return false;
+        }
 
-  if (documentSubset.hasOwnProperty(fieldName) && typeof documentSubset[fieldName] !== 'undefined' && documentSubset[fieldName] !== null) {
-    if (field.multivalued.value) {
-      if (!Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
-        return false;
-      }
+        if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
+          manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
+          return false;
+        }
 
-      if (field.multivalued.hasOwnProperty('minCount') && documentSubset[fieldName].length < field.multivalued.minCount) {
-        manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
-        return false;
+        fieldValues = documentSubset[fieldName];
       }
+      else {
+        if (Array.isArray(documentSubset[fieldName])) {
+          manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
+          return false;
+        }
 
-      if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
-        manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
-        return false;
+        fieldValues = [documentSubset[fieldName]];
       }
 
-      fieldValues = documentSubset[fieldName];
-    }
-    else {
-      if (Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
-        return false;
+      if (this.types[field.type].allowChildren) {
+        nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
       }
 
-      fieldValues = [documentSubset[fieldName]];
-    }
+      for (i = 0; i < fieldValues.length; i++) {
+        fieldErrors = [];
 
-    if (this.types[field.type].allowChildren) {
-      nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
-    }
-
-    for (i = 0; i < fieldValues.length; i++) {
-      fieldErrors = [];
+        if (!this.types[field.type].validate(field.typeOptions, fieldValues[i], fieldErrors)) {
+          if (fieldErrors.length === 0) {
+            // We still want to trigger an error, even if no message is provided
+            manageErrorMessage(field.path, errorMessages, 'An error has occurred during validation.', verbose);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 111 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)